app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -325,12 +327,20 @@ protected function addDiscounts( $totals, &$cartSubmissionData, $quote = null )
                 // others as negative, which is the Magento default.
                 // Using the absolute value.
                 $discountAmount = (int) abs(round($amount * 100));
-
-                $cartSubmissionData['discounts'][] = array(
+                $data = array(
                     'amount'      => $discountAmount,
                     'description' => $totals[$discount]->getTitle(),
                     'type'        => 'fixed_amount',
                 );
+                if ($discount === 'discount' && $quote->getCouponCode()) {
+                    $coupon = Mage::getModel('salesrule/coupon')->load($quote->getCouponCode(), 'code');
+                    $rule = Mage::getModel('salesrule/rule')->load($coupon->getRuleId());
+
+                    if (strpos($totals[$discount]->getTitle(), $rule->getName()) !== false) {
```

Comment:
can you add comment explaining why and what you do here?

Reply:
no problem.